src/libopenrave/kinbody.cpp

```diff
@@ -5984,11 +5987,12 @@ void KinBody::ExtractInfo(KinBodyInfo& info)
     info._vGrabbedInfos.resize(0);
     GetGrabbedInfo(info._vGrabbedInfos);
 
-    info._transform = GetTransform();
+    // GetTransform() = info._transform * _baseLinkRelativeTransform
+    info._transform = GetTransform() * _baseLinkRelativeTransform.inverse();
 
     // in order for link transform comparision to make sense
     KinBody::KinBodyStateSaver stateSaver(shared_kinbody(), Save_LinkTransformation);
-    SetTransform(Transform());
+    SetTransform(_baseLinkRelativeTransform); // so that base link is at exactly _baseLinkRelativeTransform
```

Comment:
why not keep it as `Transform()`?

Reply:
Originally, I was trying to preserve `GetLinks()[0]->GetTransform() == GetTransform()`. So I needed to put the body temporarily at `base link in body frame` transform, such that the `GetLinks()[0]->GetTransform()` would then be equal to the original `base link in body frame` transform when things are not changed.